lib/country_leader_board.py

```diff
@@ -3,12 +3,100 @@
 import requests
 
 
-def refresh_country_leader_board(conn, K8S_DOMAIN):
+def refresh_country_leader_board(conn):
   # check env
-  if K8S_DOMAIN is None:
-    raise Exception("K8S_DOMAIN is not defined")
+  continents = [
+    'Global',
+    'Africa',
+    'Americas',
+    'Asia',
+    'Oceania',
+  ]
+  # Limit the number of rows returned
+  limit = 10
+  # put data in dictinary
+  continenctsResult = {}
+
+  for continent in continents:
+    print ("continent:", continent)
+    # get the data from the API
+    if continent == 'Global':
+        # SQL Query pulls the number of trees planted by region, region id, region name, and centroid
+        # Example: (182490, 6632405, 'Sierra Leone', '{"type":"Point","coordinates":[-11.7927124667898,8.56329593037589]}')
+        sql = '''select r2.*,region.name, ST_AsGeoJSON(centroid) as centroid from (
+        select count(r.region_id) as planted,r.region_id as region_id from (
+        select distinct(tree_id),region_id  from public.active_tree_region
+        left join region_type on type_id = region_type.id
+        where region_type.type = 'country') r
+        group by region_id
+        order by planted desc
+        limit {}) r2
+        left join region
+        on r2.region_id = region.id'''.format(limit)
+    else:
+        sql = '''
+          select r.*, region.name, region.centroid from (
+          select count(distinct(tree_id)) planted, region_id from active_tree_region
+          where region_id in (
+          --- select records that it's centroid is within the polygon
+          select region.id from region 
+          join (select id,name,geom from region where type_id in (select id from region_type where type = 'continents') and name = '{}') continent
+          on ST_Contains(continent.geom, region.centroid)
+          where type_id in (select id from region_type where type = 'country')
+          )
+          group by region_id
+          order by planted desc
+          limit {}
+          ) r 
+          left join region on r.region_id = region.id
+        '''.format(continent,limit)
```

Comment:
@Mohmn I can not figure out how to bring you into the prod db, I made a try on this, and migrate the SQL to airflow, so in the query API, we will simply read the result from this SQL

Please help me review the second SQL for contients.

Reply:
@CareyAltgilbers I modified your code to a better structure for utilizing, please also help review, thanks for you good job on this task!